C/include/c4Replicator.h

```diff
@@ -80,6 +80,13 @@ extern "C" {
         kC4Suspended     = 0x4          ///< If true, will not connect until unsuspended
     };
 
+    /** An enumeration of the levels of progress callbacks the replicator can provide */
+    typedef C4_ENUM(int32_t, C4ReplicatorProgressLevel) {
+        kC4ReplProgressOverall,         ///< Callback about completion and estimated total
```

Comment:
Is this exactly two callbacks, one near the start that is a estimate and one after completion?

Reply:
No, it's the "running complete and total" counts callback that you get which periodically updates the states of how much progress has been made.

Reply:
Ok.  We need some kind of indicator, then (... and I think this is a good place for it) of the relative costs of each of these things.

Reply:
Cost in terms of what?  Time?  Complexity?

Reply:
I have no idea.  That's what I'm asking for.  You need to say what makes them different and why someone would choose one instead of the other.

Reply:
Ok I've added that